cmd/juju/application/storage.go

```diff
@@ -0,0 +1,267 @@
+print it in the ` + "`json`" + ` or ` + "`yaml`" + ` format using the ` + "`--format`" + ` flag:
+
+   	juju application-storage <application> --format json
+    juju application-storage <application> --format yaml
+
+To view the value of a single storage key:
+
+    juju application-storage <application> <storage-key>
+
+To set storage constraint values on an application:
+
+    juju application-storage <application> key1=size1, key2=val2 ...
+`
+	appStorageConfigExamples = `
+Print the storage constraints for all keys of the postgresql application:
+
+    juju application-storage postgresql
+
+Print the storage constraints for the pgdata storage key of the postgresql application:
+
+    juju application-storage postgresql pgdata
+
+Set the size to 100GiB, pool name to "rootfs", and count to 1 for the mysql application:
+
+    juju application-storage mysql database=100G,rootfs,1
+
+If no size is provided, Juju uses the minimum size required by the charm. If the charm does not specify a minimum, the default is 1 GiB. 
+This value is then applied when updating the application’s storage.
+
+    juju application-storage mysql database=,rootfs,1
+
+If no pool is provided, Juju selects the default storage pool from the environment.
+This pool will be recorded as the updated value for the application’s storage.
+
+	juju application-storage mysql database=100G,,1
+
+If no count is provided, Juju uses the minimum count required by the charm. 
+That count will be used when updating the application’s storage.
+
+	juju application-storage mysql database=100G,rootfs,
+`
+)
+
+var applicationStorageConfigBase = config.ConfigCommandBase{
+	Resettable: false,
+}
+
+// NewStorageCommand wraps configCommand with sane application settings.
+func NewStorageCommand() cmd.Command {
+	return modelcmd.Wrap(&storageConfigCommand{configBase: applicationStorageConfigBase})
+}
+
+type storageConfigCommand struct {
+	modelcmd.ModelCommandBase
+	configBase config.ConfigCommandBase
+	out        cmd.Output
+
+	api             StorageConstraintsAPI
+	applicationName string
+}
+
+// Info implements part of the cmd.Command interface.
+func (c *storageConfigCommand) Info() *cmd.Info {
+	info := &cmd.Info{
+		Name:     "application-storage",
+		Args:     "<application-name> [<storage-key>[=<size>,<pool>,<count>]] ...",
+		Purpose:  appStorageConfigSummary,
+		Examples: appStorageConfigExamples,
+		Doc:      appStorageConfigDoc,
+		SeeAlso: []string{
+			"storage",
+			"storage-pools",
+			"add-unit",
+		},
+	}
+
+	return jujucmd.Info(info)
+}
+
+// SetFlags implements part of the cmd.Command interface.
+func (c *storageConfigCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	c.configBase.SetFlags(f)
+
+	c.out.AddFlags(f, "tabular", map[string]cmd.Formatter{
+		"json":    cmd.FormatJson,
+		"tabular": formatConfigTabular,
+		"yaml":    cmd.FormatYaml,
+	})
+}
+
+// Init implements part of the cmd.Command interface.
```

Comment:
Same comment as above. Tell me what the end result of the init is and the state for which i find this struct.